app/controllers/spree/api/v2/storefront/account/product_questions_answers_controller.rb

```diff
@@ -14,7 +12,11 @@ def index
             private
 
             def resource
-              resource_finder.user_product_questions(spree_current_user.id)
+              if spree_current_user
+                resource_finder.for_authorized_user(spree_current_user.id)
+              else
+                resource_finder.for_unauthorized_user
+              end
```

Comment:
instead of if else in controller, I prefer fat model and thin controller and handle this logic in model
eg: 
```
 def resource
              @resource ||= resource_finder.for_user(spree_current_user, order_params)
            end
```
This also uses memoization to ensure resource_finder.for_authorized_user or resource_finder.for_unauthorized_user is called only once per request.

Let me know your thoughts though